src/features/vote/ActiveRequestsForm.tsx

```diff
@@ -1,92 +1,650 @@
+import {
+  postCommitVotes,
+  // PostRevealData,
+  // revealVotes,
+} from "web3/postVotingContractMethods";
+// import { ethers } from "ethers";
+// import toWeiSafe from "common/utils/web3/convertToWeiSafely";
+// import web3 from "web3";
+// import { computeVoteHashAncillary, Request } from "common/tempUmaFunctions";
+
+import { useCurrentRoundId } from "hooks";
+import { OnboardContext } from "common/context/OnboardContext";
+import { formatVoteDataToCommit } from "./helpers";
+import { EncryptedVote } from "web3/queryVotingContractEvents";
+
+export type FormData = {
+  [key: string]: string;
+};
+
+interface Summary {
+  identifier: string;
+  value: string;
+}
 
 interface Props {
-  activeRequests: PriceRound[];
+  activeRequests: PendingRequest[];
+  isConnected: boolean;
+  publicKey: string;
+  votePhase: string;
+  encryptedVotes: EncryptedVote[];
+  refetchEncryptedVotes: Function;
 }
 
-const ActiveRequestsForm: FC<Props> = ({ activeRequests }) => {
+interface TableValue {
+  ancillaryData: string;
+  identifier: string;
+  vote: string;
+}
+
+const ActiveRequestsForm: FC<Props> = ({
+  activeRequests,
+  isConnected,
+  publicKey,
+  votePhase,
+  encryptedVotes,
+  refetchEncryptedVotes,
+}) => {
+  const [modalState, setModalState] = useState<
+    "init" | "pending" | "success" | "error"
+  >("init");
+
+  const {
+    state: { address, network, signer },
+  } = useContext(OnboardContext);
+  const { votingContract } = useVotingContract(signer, isConnected, network);
+  // const { data } = useVotesCommittedEvents(votingContract, address);
+
+  const [tableValues, setTableValues] = useState<TableValue[]>([]);
+
+  const { data: roundId } = useCurrentRoundId();
+  const { isOpen, open, close, modalRef } = useModal();
+
+  const generateDefaultValues = useCallback(() => {
+    const dv = {} as FormData;
+    activeRequests.forEach((el) => {
+      dv[el.identifier] = "";
+    });
+
+    return dv;
+  }, [activeRequests]);
+
+  const { handleSubmit, control, watch, reset } = useForm<FormData>({
+    defaultValues: generateDefaultValues(),
+  });
+
+  const onSubmit = useCallback(
+    (data: FormData) => {
+      const validValues = {} as FormData;
+      for (let i = 0; i < Object.keys(data).length; i++) {
+        if (Object.values(data)[i] !== "")
+          validValues[Object.keys(data)[i]] = Object.values(data)[i];
+      }
+
+      // Format data.
+      formatVoteDataToCommit(
+        validValues,
+        activeRequests,
+        roundId,
+        address,
+        publicKey
+      ).then((fd) => {
+        if (votingContract) {
+          postCommitVotes(votingContract, fd).then((tx) => {
+            setModalState("pending");
+            // Need to confirm if the user submits the vote.
+            if (tx) {
+              tx.wait(1).then((conf: any) => {
+                // Temporary, as mining is instant on local ganache.
+                setTimeout(() => setModalState("success"), 5000);
```

Comment:
might want to wrap this in a check for network id or a test env so we dont have to remember to pull this out